### ISSUE TITLE
IntendedFor field in json does not have 'run-0X' in filename

### ISSUE THREAD

**Issue by yonestar** [jmtyszka/bidskit#33]
Hi Mike,

Since I know you are working on all this stuff now, wanted to add one more thing to the mix.  In the json files accompanying the field maps, the IntendedFor field points to the functional run names -- but without with run-0X "suffix".

In other words:  current version of bidskit appends run-0X to all functional runs (which is OK/BIDS-compliant), but the functional filenames in the IntendedFor fields do not include the run-0X, and thus point to a non-existent file.

Thank you thank you!
Yoni

**Comment by jmtyszka** [comment]
Hi Yoni,

Sorry for the delay in replying to this issue. Automatic generation of run numbers for IntendedFor filenames is actually quite difficult, since there are several approaches people use to assigning fieldmaps to EPI series. Off the top of my head:

1. One fieldmap for all EPI series in an experiment
2. One fieldmap per EPI series, acquired either immediately before or after the EPI series.
3. One fieldmap between each EPI series (ie a total of n-1 fieldmaps for n EPI series)
4. One fieldmap at the beginning and one fieldmap at the end of the experiment.

There are probably other schemes out there, but in the end it means that only the experimenter really knows when fieldmap to attach to which EPI series. If you're using the same series description for all EPI series, then the run-XX key will be added to the filename. If you're also using identical series descriptions  for all fieldmap acquisitions, I can't immediately think of a way to sort out which fieldmap to assign to which EPI series because of the variations in fieldmap ordering above.

Maybe proximity in time of the fieldmap to the start of the EPI series?

Let me know what you think the expected behavior should be and I'll look into implementations.

**Comment by yonestar** [comment]
Hi Mike,

I think my first message may have been unclear -- or perhaps I am misunderstanding your reply.  But regardless, thinking more about my original issue, I have a solution that doesn't require any edits to the code.

Originally, my ProtocolTranslator.json for field map run looked like:
```
    "bladder_distortion_corr_32ch_pa_polarity_invert_to_ap":[
        "fmap",
        "dir-ap_bladder_epi",
        "task-bladder_bold"
    ],
```

The issue was that the functional files were coming out as `task-bladder_bold-run-01` or `task-bladder_bold-run-02`, while the field map's json file had only `IntendedFor: task-bladder_bold`  -- noticed the names don't match b/c of the `run-0X`.

My solution is to edit the ProtocolTranslator.json to have: 
```
    "bladder_distortion_corr_32ch_pa_polarity_invert_to_ap":[
        "fmap",
        "dir-ap_bladder_epi",
        ["task-bladder_bold", "task-bladder_bold-run01", "task-bladder_bold-run02", "task-bladder_bold-run03"]
    ],
```
which will cover all my bases.  Even though I intend to have only 1 task-bladder run, if there was an error in the run, I end up with two or maybe three sets of dicoms if things got really screwy during the scan.  The above sample ProtocolTranslator covers all those cases.

Long story short, I've thought through it more and there is no issue!

Thanks
Yoni
